udfs/migration/snowflake/factorial.sql

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+-- factorial:
+-- Computes the factorial of its input. The input argument must be an integer expression in the range of 0 to 27. 
+-- Due to data type differences, the maximum input value in BigQuery is smaller than in Snowflake. 
+-- Input:
+-- integer_expr: INT64
+-- Output: NUMERIC
```

Comment:
Looks like the Snowflake factorial function returns an integer, is there a reason we return NUMERIC?

Reply:
If we use INT64, the max returned value would be `20!`. The larger INT64 values produce an overflow error. With NUMERIC the max value is `27!` which is a bit better (however still smaller than Snowflake's `33!`).

Reply:
Got it. 

nit: Can we add a trailing 0 to each value. Example: `1.0`

Reply:
Added trailing zeros.